### PR TITLE
Migrate /vip to Helix API

### PR DESCRIPTION
**Pull request by aidenwallis**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

Handles #3983 

Migrates `/vip` to Helix API. The majority of the errors returned by the helix endpoint are actually pretty good, and are more or less the same as what are returned by IRC, so I just forwarded them for most status codes.

### CODE REVIEW

File: src/providers/twitch/api/Helix.cpp [hunk 1]
```diff
@@ -1178,6 +1178,87 @@ void Helix::sendChatAnnouncement(
...
+    urlQuery.addQueryItem("broadcaster_id", broadcasterID);
+    urlQuery.addQueryItem("user_id", userID);
+
+    this->makeRequest("channels/vips", urlQuery)
+        .type(NetworkRequestType::Post)
+        .onSuccess([successCallback, failureCallback](auto result) -> Outcome {
+            if (result.status() != 204)
+            {
+                qCWarning(chatterinoTwitch)
+                    << "Success result for deleting chat messages was"
```
Comment (iProdigy) [inline src/providers/twitch/api/Helix.cpp:1198]:
```suggestion
                    << "Success result for adding a VIP was"
```

Comment (aidenwallis) [reply src/providers/twitch/api/Helix.cpp:1198]: Yes - it's still a draft, this is only up to help with diffing some stuff

**Review by iProdigy — APPROVED** [review]
LGTM

File: tests/src/HighlightController.cpp [hunk 2]
```diff
@@ -225,6 +225,12 @@ class MockHelix : public IHelix
                       failureCallback)),
                 (override));
 
+    // The extra parenthesis around the failure callback is because its type contains a comma
+    MOCK_METHOD(void, addChannelVIP,
+                (QString broadcasterID, QString userID,
+                 ResultCallback<> successCallback,
+                 (FailureCallback<HelixAddChannelVIPError, QString>
```
Comment (iProdigy) [inline tests/src/HighlightController.cpp:232]:
```suggestion
                 (FailureCallback<HelixAddChannelVIPError, QString>
                      failureCallback)),
                (override));
```

**Review by pajlada — APPROVED** [review]

**Comment by pajlada** [comment]
Thank you @aidenwallis !

**Pull request merged**